internal/service/substrate.go

```diff
@@ -46,8 +46,8 @@ type SubscribeService struct {
 func (s *Service) initSubscribeService(done chan struct{}) *SubscribeService {
 	return &SubscribeService{
 		Service:    s,
-		newHead:    make(chan bool, 1),
-		newFinHead: make(chan int, 1),
+		newHead:    make(chan bool, 1000000),
+		newFinHead: make(chan int, 1000000),
```

Comment:
Why 1 million?

Reply:
TLDR; frantic debugging. Started getting hangs with new finalized blocks not being received because when fetching "catch up" blocks `newFinHead` doesn't get received - so there was just a big build up of waiting senders.

I was like "well what if it just never blocked" which led to this. I pushed a commit to try to be a bit smarter here (if the last `newFinHead` hasn't been received, just receive it and replace it with the new value)